core/src/main/java/io/questdb/griffin/engine/groupby/SampleByInterpolateRecordCursorFactory.java

```diff
@@ -142,13 +164,47 @@ public SampleByInterpolateRecordCursorFactory(
                     interpolatorFunctions.add(InterpolationUtil.INTERPOLATE_FLOAT);
                     break;
                 default:
-                    Misc.freeObjList(groupByFunctions);
+                    Misc.freeObjList(groupByScalarFunctions);
+                    throw SqlException.$(function.getPosition(), "Unsupported type: ").put(ColumnType.nameOf(function.getType()));
+            }
+        }
+
+        this.groupByTwoPointFunctionCount = groupByTwoPointFunctions.size();
+        for (int i = 0; i < groupByTwoPointFunctionCount; i++) {
+            GroupByFunction function = groupByTwoPointFunctions.getQuick(i);
+            switch (function.getType()) {
+                case ColumnType.BYTE:
+                    interpolatorGapTwoPointFunctions.add(InterpolationUtil.INTERPOLATE_GAP_BYTE);
```

Comment:
The "two point" function interpolation API is the same for every result type.

What is the point of having collections of function references here?

Reply:
fixed